app/gui/src/project-view/components/GraphEditor/widgets/WidgetText.vue

```diff
@@ -141,35 +179,43 @@ export const widgetDefinition = defineWidget(
   <label
     ref="widgetRoot"
     class="WidgetText widgetRounded"
+    :class="{ singleLine: !isMultiline }"
     @pointerdown.stop.prevent="focusEditor"
     @click.stop
   >
-    <NodeWidget v-if="openToken" :input="WidgetInput.FromAst(openToken)" />
+    <NodeWidget v-if="openToken" :input="WidgetInput.FromAst(openToken)" class="delimiter open" />
     <!-- Do not finish edit on blur here!
 
     It is tempting, but it breaks the cooperation with possible drop-down widget. Blur may be done on
     pointerdown, and if it would end the interaction, the drop down would also be hidden, making 
     any `click` event on it impossible.
     -->
-    <CodeMirrorInlineRoot
+    <CodeMirrorRoot
       ref="editorRoot"
-      @keydown.enter.stop="accepted"
-      @keydown.tab.stop="accepted"
+      :data-syntax-language="syntaxLanguage"
       @focusin="editing.start()"
+      @keydown.enter.capture="onEnter"
+      @keydown.tab.stop.capture="accepted"
+      @keydown.up.stop
+      @keydown.down.stop
+    />
```

Comment:
We may consider putting this into `ResizableWidget` - this not only adds resize handles, but also synchronizes with node size if it's the only resizable element - so node consisting of `'''` will behave as Table.input nodes.

But not necessary to merge. If non-trivial, let's make a separate issue.

Reply:
Currently the size fits the content, up to a limit. The ideal behaviour may be a combination of auto-sizing and manual resizability, but